File: app/src/main/java/com/u1fukui/bbs/view/customview/ErrorView.java
package com.u1fukui.bbs.view.customview;

import android.content.Context;
import android.databinding.BindingAdapter;
import android.databinding.DataBindingUtil;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.u1fukui.bbs.R;

public class ErrorView extends LinearLayout {

    public interface ErrorViewListener {
        void onClickReloadButton();
    }

    public ErrorView(Context context) {
        super(context);
        init();
    }

    public ErrorView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public ErrorView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        setOrientation(LinearLayout.VERTICAL);
        setGravity(Gravity.CENTER);
        DataBindingUtil.inflate(LayoutInflater.from(getContext()), R.layout.view_error, this, true);
    }

    @BindingAdapter("errorMessage")
    public static void errorMessage(View view, String message) {
        TextView messageView = (TextView) view.findViewById(R.id.error_message);
        messageView.setText(message);
    }

    @BindingAdapter("errorViewListener")
    public static void onClickReloadButton(View view, final ErrorViewListener listener) {
        View reloadButton = view.findViewById(R.id.reload_btn);
        if (reloadButton != null && listener != null) {
            reloadButton.setOnClickListener(new OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (listener != null) {
                        listener.onClickReloadButton();
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/u1fukui/bbs/customview/ErrorView.kt
package com.u1fukui.bbs.customview

import android.content.Context
import androidx.databinding.BindingAdapter
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import android.util.AttributeSet
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView

import com.u1fukui.bbs.R

class ErrorView @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null,
        defStyleAttr: Int = 0
) : LinearLayout(context, attrs, defStyleAttr) {

    interface ErrorViewListener {
        fun onClickReloadButton()
    }

    init {
        orientation = LinearLayout.VERTICAL
        gravity = Gravity.CENTER
        DataBindingUtil.inflate<ViewDataBinding>(LayoutInflater.from(context), R.layout.view_error, this, true)
    }

    companion object {

        @BindingAdapter("errorMessage")
        @JvmStatic
        fun errorMessage(view: View, message: String?) {
            val messageView = view.findViewById<View>(R.id.error_message) as TextView
            messageView.text = message
        }

        @BindingAdapter("errorViewListener")
        @JvmStatic
        fun onClickReloadButton(view: View, listener: ErrorViewListener?) {
            view.findViewById<View>(R.id.reload_btn).apply {
                setOnClickListener {
                    listener?.onClickReloadButton()
                }
            }
        }
    }
}
